test/integration/api/submissions.js

```diff
@@ -1399,6 +1399,103 @@ describe('api: /forms/:id/submissions', () => {
             .then(({ body }) => { body.reviewState.should.equal('approved'); })))));
   });
 
+  describe('/:instanceId DELETE', () => {
+    it('should reject notfound if the submission does not exist', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.delete('/v1/projects/1/forms/simple/submissions/one')
+          .expect(404))));
+
+    it('should reject if the user cannot delete', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.post('/v1/projects/1/forms/simple/submissions')
+          .send(testData.instances.simple.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)
+          .then(() => service.login('chelsea', (asChelsea) =>
+            asChelsea.delete('/v1/projects/1/forms/simple/submissions/one')
+              .expect(403))))));
+
+    it('should soft-delete the submission and not be able to access it again', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.post('/v1/projects/1/forms/simple/submissions')
+          .send(testData.instances.simple.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)
+          .then(() => asAlice.delete('/v1/projects/1/forms/simple/submissions/one')
+            .expect(200))
+          .then(() => asAlice.get('/v1/projects/1/forms/simple/submissions/one')
+            .expect(404)))));
+
+    it('should not let a draft submission be deleted', testService(async (service) => {
+      const asAlice = await service.login('alice');
+      await asAlice.post('/v1/projects/1/forms/simple/draft');
+      await asAlice.post('/v1/projects/1/forms/simple/draft/submissions')
+        .send(testData.instances.simple.one)
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+      // draft submission delete resource does not exist
+      await asAlice.delete('/v1/projects/1/forms/simple/draft/submissions/one')
+        .expect(404);
+    }));
```

Comment:
You can probably have a draft submission with the same instance id as a deleted non-draft submission. Does that have any weird implications? Should some queries check for draft or is it ok because they check deletedAt!=null and you can never delete a draft?